modules/gojira-core/src/main/java/com/flipkart/gojira/core/ProfileRepository.java

```diff
@@ -56,6 +56,20 @@ protected String initialValue() {
         }
       };
   private static ProfileSetting globalProfileSetting = new ProfileSetting();
+
+  private static InheritableThreadLocal<ProfileSetting> requestProfileSetting =
```

Comment:
I don't think introducing a new thread local is a good idea. Please remove this. We can make requestLevelSetting to an attribute within ProfileData.

Maybe rename ProfileData -> ExecuteData. It already stores TestData and ProfileState. This looks like the correct place for ProfileSetting as well. We can continue to have a global variable that tracks the jvm level setting and additionally set the value inside ProfileData/ExecuteData and store the current profileSetting independent of the jvm level setting.

Reply:
Done